src/docfx/lib/log/Log.cs

```diff
@@ -0,0 +1,142 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Immutable;
+using System.Diagnostics;
+using System.Threading;
+
+namespace Microsoft.Docs.Build
+{
+    internal static class Log
```

Comment:
>Log [](start = 26, length = 3)

can we save the logs to build output for our debuging?

Reply:
or accept more listener

---
In reply to: [198065820](https://github.com/dotnet/docfx/pull/2942#discussion_r198065820) [](ancestors = 198065820)

Reply:
yes, it will be in later PR, this PR focuses on progress.

---
In reply to: [198065909](https://github.com/dotnet/docfx/pull/2942#discussion_r198065909) [](ancestors = 198065909,198065820)